libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/utils/distance/MapboxDistanceFormatter.kt

```diff
@@ -0,0 +1,158 @@
+package com.mapbox.navigation.trip.notification.utils.distance
+
+import android.content.Context
+import android.graphics.Typeface
+import android.text.SpannableString
+import android.text.Spanned
+import android.text.style.RelativeSizeSpan
+import android.text.style.StyleSpan
+import com.mapbox.navigation.base.formatter.DistanceFormatter
+import com.mapbox.navigation.base.typedef.IMPERIAL
+import com.mapbox.navigation.base.typedef.METRIC
+import com.mapbox.navigation.base.typedef.RoundingIncrement
+import com.mapbox.navigation.base.typedef.VoiceUnit
+import com.mapbox.navigation.trip.notification.R
+import com.mapbox.navigation.trip.notification.utils.extension.getUnitTypeForLocale
+import com.mapbox.navigation.utils.extensions.inferDeviceLocale
+import com.mapbox.turf.TurfConstants.UNIT_FEET
+import com.mapbox.turf.TurfConstants.UNIT_KILOMETERS
+import com.mapbox.turf.TurfConstants.UNIT_METERS
+import com.mapbox.turf.TurfConstants.UNIT_MILES
+import com.mapbox.turf.TurfConversion
+import java.text.NumberFormat
+import java.util.Locale
+
+/**
+ * Creates an instance of DistanceFormatter, which can format distances in meters
+ * based on a language format and unit type.
+ *
+ *
+ * This constructor will infer device language and unit type using the device locale.
+ *
+ * @param language for which language
+ * @param unitType to use, or NONE_SPECIFIED to use default for locale country
+ * @param context from which to get localized strings from
+ * @param roundingIncrement increment by which to round small distances
+ */
+class MapboxDistanceFormatter(
```

Comment:
This class will need to live in libnavigation-core because we require it in the constructor.